brainscore_language/plugins/futrell2018/data_packaging.py

```diff
@@ -0,0 +1,153 @@
+    WorkTimeInSeconds_meta = [v[1] for v in metaInfo_subjects.values()]
+
+    # get metadata for presentation dimension
+    word_df = pd.read_csv(stories_file, sep='\t')
+    voc_item_ID = np.array(word_df['item'])
+    voc_zone_ID = np.array(word_df['zone'])
+    voc_word = np.array(word_df['word'])
+
+    # get sentence_IDs (finds 481 sentences)
+    sentence_ID = []
+    idx = 1
+    for i, elm in enumerate(voc_word):
+        sentence_ID.append(idx)
+        if elm.endswith((".", "?", "!", ".'", "?'", "!'", ";'")):
+            if i + 1 < len(voc_word):
+                if not (voc_word[i + 1].islower() or voc_word[i] == "Mr."):
+                    idx += 1
+
+    # get IDs of words within a sentence
+    word_within_a_sentence_ID = []
+    idx = 0
+    for i, elm in enumerate(voc_word):
+        idx += 1
+        word_within_a_sentence_ID.append(idx)
+        if elm.endswith((".", "?", "!", ".'", "?'", "!'", ";'")):
+            if i + 1 < len(voc_word):
+                if not (voc_word[i + 1].islower() or voc_word[i] == "Mr."):
+                    idx = 0
+            else:
+                idx = 0
+
+    # stimulus_ID
+    stimulus_ID = list(range(1, len(voc_word) + 1))
+
+    # add word_core that treats e.g. "\This" and "This" as the same words (to split over)
+    word_core = [re.sub(r'[^\w\s]', '', word) for word in voc_word]
+
+    # build xarray
+    # voc_word = word
+    # voc_item_ID = index of story (1-10)
+    # voc_zone_ID = index of words within a story
+    # sentence_ID = index of words within each sentence
+    # stimulus_ID = unique index of word across all stories
+    # subjects = WorkerIDs
+    # correct_meta = number of correct answers in comprehension questions
+    assembly = BehavioralAssembly(reading_times,
+                                  dims=('presentation', 'subject'),
+                                  coords={'word': ('presentation', voc_word),
+                                          'word_core': ('presentation', word_core),
+                                          'story_id': ('presentation', voc_item_ID),
+                                          'word_id': ('presentation', voc_zone_ID),
+                                          'word_within_sentence_id': ('presentation', word_within_a_sentence_ID),
+                                          'sentence_id': ('presentation', sentence_ID),
+                                          'stimulus_id': ('presentation', stimulus_ID),
+                                          'subject_id': ('subject', subjects),
+                                          'correct': ('subject', correct_meta),
+                                          'WorkTimeInSeconds': ('subject', WorkTimeInSeconds_meta),
+                                          })
+
+    # upload
+    upload_data_assembly(assembly,
+                         assembly_identifier="Futrell2018",
+                         bucket_name="brainscore-language")
```

Comment:
is this data the previous PNAS version? if so, I think they discovered some minor bug in their alignment, and uploaded a corrected version recently https://github.com/languageMIT/naturalstories

Reply:
it is from PNAS yes -- I would put this as a todo for the next version